gninasrc/gninavis/cnn_visualization.cpp

```diff
@@ -1080,28 +967,56 @@ std::vector<float> cnn_visualization::remove_fragments(int size) {
     int counter = 1; //stores current fragment number for progress output
 
     for (auto path = paths.begin(); path != paths.end(); ++path) //iterate through path lengths
-            {
+    {
         std::list<std::vector<int>> list_of_lists = std::get < 1 > (*path);
 
         for (auto bonds = list_of_lists.begin(); bonds != list_of_lists.end();
                 ++bonds) //iterate through paths of a given length
-                {
+        {
             std::vector<int> bond_list = *bonds;
 
-            if (!visopts.verbose) {
+            if (!visopts.verbose)
+            {
                 std::cout << "Scoring fragments: " << counter << '/'
                         << path_count << '\r' << std::flush;
                 counter++;
             }
 
             for (int i = 0; i < bond_list.size(); ++i) //iterate through bonds in path
-                    {
+            {
                 RDKit::Bond bond = *(rdkit_mol.getBondWithIdx(bond_list[i]));
-                int first = bond.getBeginAtomIdx();
-                int second = bond.getEndAtomIdx();
-
-                atoms_to_remove.insert(first + 1);
-                atoms_to_remove.insert(second + 1);
+                int first_index = bond.getBeginAtomIdx();
+                int second_index = bond.getEndAtomIdx();
+                RDGeom::Point3D first_coords = conf.getAtomPos(first_index);
+                RDGeom::Point3D second_coords = conf.getAtomPos(second_index);
+
+                //round to same coordinate precision as pdb
+                std::stringstream ss;
+                ss << std::fixed << std::setprecision(3) << first_coords.x;
+                std::string rounded_x = ss.str();
+                ss.str("");
+                ss << std::fixed << std::setprecision(3) << first_coords.y;
+                std::string rounded_y = ss.str();
+                ss.str("");
+                ss << std::fixed << std::setprecision(3) << first_coords.z;
+                std::string rounded_z = ss.str();
+
+                std::string first_xyz = rounded_x + rounded_y + rounded_z;
+
+                ss.str("");
```

Comment:
since you do this operation in a number of different places it should be a helper function, e.g. rounded_num_to_str